app/src/main/java/com/benefit/adapters/ConversationAdapter.java

```diff
@@ -0,0 +1,127 @@
+            lastChatDate = itemView.findViewById(R.id.last_chat_time);
+            giveOrWantText = itemView.findViewById(R.id.give_or_want);
+            itemView.setOnClickListener(this);
+        }
+
+        void setProduct(Product product) {
+            this.product = product;
+            this.productTitle.setText(product.getTitle());
+            //set picture
+        }
+
+        void setProductOwnerName(String productOwner) {
+            this.productOwnerName.setText(productOwner);
+        }
+
+        void setLastChatDate(Date date){
+            lastChatDate.setText(dateFormat.format(date));
+        }
+
+        void setMatch(Match match){
+            this.match = match;
+        }
+
+        @Override
+        public void onClick(View v) {
+            Context context = v.getContext();
+            Intent intent = new Intent(context, ChatActivity.class);
+
+        }
+    }
+
+    private DatabaseDriver databaseDriver;
+
+    public ConversationAdapter(@NonNull FirestoreRecyclerOptions<Match> options, DatabaseDriver databaseDriver) {
+        super(options);
+        this.databaseDriver = databaseDriver;
+    }
+
+    @NonNull
+    @Override
+    public ConversationHolder onCreateViewHolder(@NonNull ViewGroup group, int viewType) {
+        View view = LayoutInflater.from(group.getContext()).inflate(R.layout.conversation, group, false);
+
+        return new ConversationHolder(view);
+    }
+
+    @Override
+    protected void onBindViewHolder(@NonNull ConversationHolder conversationHolder, int i, @NonNull Match match) {
+        conversationHolder.setMatch(match);
+        conversationHolder.setLastChatDate(match.getTimestamp());
+        databaseDriver.getCollectionReferenceByName("products")
+                .whereEqualTo("id", match.getProductId())
+                .get()
+                .addOnCompleteListener(task -> {
+                    if(task.isSuccessful()){
+                        for (QueryDocumentSnapshot document : Objects.requireNonNull(task.getResult())) {
+                            conversationHolder.setProduct(document.toObject(Product.class));
+                        }
+                    }
+                });
+        if (match.getSellerId().equals(databaseDriver.getAuth().getUid())){
+            conversationHolder.setProductOwnerName("your item");
+        }
+        else {
+            databaseDriver.getCollectionReferenceByName("users")
+                    .whereEqualTo("uid", match.getSellerId())
+                    .get()
+                    .addOnCompleteListener(task -> {
+                       if (task.isSuccessful()){
+                           for (QueryDocumentSnapshot document : Objects.requireNonNull(task.getResult())) {
+                               User sellerUser = document.toObject(User.class);
+                               conversationHolder.setProductOwnerName(sellerUser.getFirstName() + " " + sellerUser.getLastName());
```

Comment:
I think it will be better if we'll put in the User object a method getFullName which returns sellerUser.getFirstName() + " " + sellerUser.getLastName().
It's more clean, it might be needed in other classes, and it will be easier to change in future (for exxample if we want to seperate with comma and not space).

Reply:
great idea!